packages/fetch/src/fetch.client.ts

```diff
@@ -56,7 +69,7 @@ export class FetchClient<Opts extends FetchOptions<any, any> = RequestInit, Req
 
   private createMethod(method: HTTPMethod) {
     return (url: string | Req, options?: Omit<RequestOptions<Opts>, 'method'>) => {
-      const response = this.send(url, { ...options, method } as RequestOptions<Opts>);
+      const response = this.send(url, { ...options, method } as unknown as RequestOptions<Opts>);
```

Comment:
I tested it locally and I think we can avoid casting to `unknown` here first  if we just add ` | undefined` to the `ExtraOptions.searchParams` type

It seems to be complaining because of this fancy new typescript rule we have enabled
https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes